components/ServerForm.tsx

```diff
@@ -55,10 +85,58 @@ export default function ServerForm({
         inputMode="url"
         keyboardType="url"
         autoCapitalize="none"
-        onSubmitEditing={validateAndSaveURL}
-        returnKeyType="go"
+        onSubmitEditing={() =>
+          basicAuth.basicAuthRequired
+            ? usernameRef.current.focus()
+            : validateAndSaveURL()
+        }
+        returnKeyType={basicAuth.basicAuthRequired ? "next" : "go"}
         autoCorrect={false}
       />
+      <CheckBox
+        checked={basicAuth.basicAuthRequired}
+        onChange={(nextValue) => setBasicAuthRequired(nextValue)}
+      >
+        Anmeldung erforderlich
+      </CheckBox>
+      {basicAuth.basicAuthRequired && (
+        <Input
+          style={{ marginTop: 5 }}
+          size="large"
+          status={validateUsername() ? "basic" : "danger"}
+          onChangeText={(nextValue) => setBasicAuthUserName(nextValue)}
+          value={basicAuth.username}
+          inputMode="text"
+          keyboardType="default"
+          autoCapitalize="none"
+          returnKeyType="next"
+          autoCorrect={false}
+          placeholder="Benutzer"
+          label="Benutzer"
+          ref={usernameRef}
+          onSubmitEditing={() => passwordRef.current.focus()}
+        />
+      )}
+      {basicAuth.basicAuthRequired && (
+        <Input
+          style={{ marginTop: 5 }}
```

Comment:
I've adjusted the spaced a bit locally. We have enough vertical space to make this less condense.

![Simulator Screenshot - iPhone 15 - 2024-08-19 at 20 53 40](https://github.com/user-attachments/assets/5b577e19-0b6d-4b18-8aab-d413912cb347)

![Simulator Screenshot - iPhone 15 - 2024-08-19 at 20 53 43](https://github.com/user-attachments/assets/438e2cd1-b6d0-40bc-be97-cbd0f4d33b44)